include/caffe/syncedmem.hpp

```diff
@@ -78,6 +78,7 @@ class SyncedMemory {
   bool cpu_malloc_use_cuda_;
   bool own_gpu_data_;
   int gpu_device_;
+  cudaStream_t stream_;
```

Comment:
Is this necessary? I believe the goal here is to be aware of the current stream we are executing on and send this info along to stream-dependant methods (like CUB allioc/free), but not to create new streams or retain any stream info.